### PR TITLE
Update Prow to the latest versions used by k8s

### PR DESCRIPTION
**Pull request by afrittoli**
# Changes

Update Prow to the latest versions used by k8s

Update various Prow and related microservices to the latest
versions used in k8s test-infra.
Reference manifests: https://github.com/kubernetes/test-infra/tree/master/config/prow

- plank has been replaced by prow controller manager
- deck includes spyglass which shall replace gubernator (not removed
  yet)
- updated all the container images
- add metric ports and services
- added ghproxy everywhere
- created a Tektoncd wide GitHub oauth app, which is used by deck.
  Documented associated process and created the secrets on the cluster.

The k8s test-infra runs boskos janitor, but only for aws and non-gke
resources, while we run it today for gke-resources only.
My impression is that it's merely a naming difference, since upstream
does not configure any gke-resource, only gce-resource. I'm leaving
the existing naming for now, but in future we may want to align to
test-infra upstream naming. I imaging the logic being:
- GCE resources: projects
- GKE resources (dynamic): clusters

The Spyglass configuration seems to be working ok, but over time we
may have to tune the highlight regexes for Tekton specific messages.

Tide has now the ability to flush some data to a GCS bucket, but I
could not figure out how to give tide access to the bucket, so I did
not configure it for now. We can do so as a follow-up.

Once this is merged, we may consider splitting the manifests similar to
what k8s test-infra does, to make it easier in future to compare and
update where necessary:
- one manifest for prow configs
- one manifest for prow jobs
- for each component, one manifest each for deployment, service, rbac

Fix a few yamllint warnings too.

Signed-off-by: Andrea Frittoli <andrea.frittoli@uk.ibm.com>

/kind misc

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: prow/config.yaml [hunk 1]
```diff
@@ -38,7 +59,24 @@ branch-protection:
...
       enforce_admins: false
+      # Enforce EasyCLA for all repos
+      required_status_checks:
+        contexts:
+        - EasyCLA
+      repos:
+        pipeline:
+          required_status_checks:
+            contexts:
+            - check-pr-has-kind-label
```
Comment (afrittoli) [inline prow/config.yaml:70]: This is nice, we can use prow to configured required tekton jobs :)

File: prow/config.yaml [hunk 2]
```diff
@@ -38,7 +59,24 @@ branch-protection:
...
+
+sinker:
+  resync_period: 1m
+  max_prowjob_age: 48h
+  max_pod_age: 48h
+  terminated_pod_ttl: 30m
```
Comment (afrittoli) [inline prow/config.yaml:76]: This seems new, I assume the values from k8s are ok for us too

File: prow/config.yaml [hunk 3]
```diff
@@ -86,6 +124,12 @@ tide:
     tektoncd/triggers: rebase
     tektoncd/website: rebase
   target_url: https://prow.tekton.dev/tide
+  priority:
+  - labels: [ "kind/flake", "priority/critical-urgent" ]
+  - labels: [ "kind/failing-test", "priority/critical-urgent" ]
+  - labels: [ "kind/bug", "priority/critical-urgent" ]
```
Comment (afrittoli) [inline prow/config.yaml:130]: Seems reasonable to have

File: prow/config.yaml [hunk 4]
```diff
@@ -90,8 +128,128 @@ tide:
     tektoncd/triggers: rebase
     tektoncd/website: rebase
   target_url: https://prow.tekton.dev/tide
-sinker:
-  max_pod_age: 3h
```
Comment (abayer) [inline prow/config.yaml:94]: This got added after you opened your PR, and we do want it. =)

File: prow/config.yaml [hunk 5]
```diff
@@ -90,8 +128,128 @@ tide:
...
+      required_files:
+        - ^artifacts/filtered\.cov$
+      optional_files:
+        - ^artifacts/filtered\.html$
+    - lens:
+        name: podinfo
+        config:
+          runner_configs:
+            "default":
+              pod_link_template: "https://console.cloud.google.com/kubernetes/pod/us-central1-f/prow/test-pods/{{ .Name }}/details?project=k8s-prow-builds"
```
Comment (abayer) [inline prow/config.yaml:241]: These links aren't valid for us, obviously, but I'm assuming you just haven't configured spyglass at all yet?

**Review by vdemeester — APPROVED** [review]

File: prow/config.yaml (hunk 5 above)
Comment (afrittoli) [reply prow/config.yaml:241]: Good catch, I did configure some of it, so all the top links work, but I had missed these :)

File: prow/config.yaml [hunk 6]
```diff
@@ -17,20 +17,41 @@ pod_namespace: test-pods
 
 plank:
   job_url_template: 'https://tekton-releases.appspot.com/build/tekton-prow/{{if or (eq .Spec.Type "presubmit") (eq .Spec.Type "batch")}}pr-logs/pull{{with .Spec.Refs}}/{{.Org}}_{{.Repo}}{{end}}{{else}}logs{{end}}{{if eq .Spec.Type "presubmit"}}/{{with index .Spec.Refs.Pulls 0}}{{.Number}}{{end}}{{else if eq .Spec.Type "batch"}}/batch{{end}}/{{.Spec.Job}}/{{.Status.BuildID}}/'
-  report_template: '[Full PR test history](https://tekton-releases.appspot.com/pr/{{.Spec.Refs.Org}}_{{.Spec.Refs.Repo}}/{{with index .Spec.Refs.Pulls 0}}{{.Number}}{{end}}). [Your PR dashboard](https://prow.tekton.dev/pr/{{with index .Spec.Refs.Pulls 0}}{{.Author}}{{end}}).'
+  report_templates:
+    '*': '[Full PR test history](https://tekton-releases.appspot.com/pr/{{.Spec.Refs.Org}}_{{.Spec.Refs.Repo}}/{{with index .Spec.Refs.Pulls 0}}{{.Number}}{{end}}). [Your PR dashboard](https://prow.tekton.dev/pr/{{with index .Spec.Refs.Pulls 0}}{{.Author}}{{end}}).'
```
Comment (afrittoli) [inline prow/config.yaml:21]: This is still gubernator, I guess we could switch it to spyglass in this PR already.

Comment (afrittoli) [reply prow/config.yaml]: Fixed now

**Pull request merged**